app/graphql/mutations/change_visibility.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: false
+
+module Mutations
+  class ChangeVisibility < BaseMutation
+    argument :id, Integer, required: true
+    argument :record_type, String, required: true
+    argument :visible, Boolean, required: true
+
+    type Types::ChangeVisibilityType
+
+    RECORD_WHITELIST = ["EventRecord", "NewsItem", "PointOfInterest", "Tour"].freeze
+
+    def resolve(id:, record_type:, visible:)
+      return error_status("recordType") unless RECORD_WHITELIST.include?(record_type)
+      return error_status("visible") unless [true, false].include?(visible)
+
+      record = record_type.constantize
+                 .filtered_for_current_user(context[:current_user])
+                 .where(id: id)
+                 .first
+      return error_status if record.blank?
+
+      record.update(visible: visible)
+
+      OpenStruct.new(id: id, status: "visibility set to #{visible}", status_code: 200)
```

Comment:
vllt zur lesbarkeit auch wie `error_status` in eine private method `success_status`?

Reply:
hatte ich erst, fand es etwas Overhead für ein einfaches, kurzes OpenStruct eine Methode mit 2 Übergabeparametern zu machen

Reply:
ok